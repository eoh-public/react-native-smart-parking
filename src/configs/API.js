import Config from 'react-native-config';

const API_ROOT = Config.API_ROOT;

const API = {
  AUTH: {
    LOGIN_PHONE: API_ROOT + '/accounts/login/',
    LOGIN_FACEBOOK: API_ROOT + '/accounts/login/facebook/',
    REGISTER_PHONE: API_ROOT + '/accounts/register/phone/',
    RESEND_OTP: API_ROOT + '/accounts/register/resend_otp/',
    VERIFY_OTP: API_ROOT + '/accounts/register/verify_otp/',
    LOGIN_SOCIAL_FB: API_ROOT + '/accounts/login/facebook/',
    LOGIN_SOCIAL_GG: API_ROOT + '/accounts/login/google/',
    LOGIN_SOCIAL_AP: API_ROOT + '/accounts/login/apple/',
    FORGOT_PASSWORD: API_ROOT + '/accounts/forgot_password/',
    FORGOT_PASSWORD_VERIFY_OTP:
      API_ROOT + '/accounts/forgot_password/verify_otp/',
    RESET_PASSWORD: API_ROOT + '/accounts/forgot_password/set_password/',
  },
  ACCOUNTS: {
    FEATURES: API_ROOT + '/accounts/features/',
    ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
    REMOVE_CARD: (id) =>
      API_ROOT + `/billing/payments/stripe/remove_cards/${id}/`,
    CHANGE_DEFAULT_CARD: API_ROOT + '/billing/payments/stripe/set_default/',
    LIST_PAYMENT_METHODS: API_ROOT + '/billing/list_payment_methods/',
    CREATE_CARD_TOKEN: 'https://api.stripe.com/v1/tokens',
  },
  UNIT: {
    ADD_USER: API_ROOT + '/property_manager/shared_units/add_by_phone_number/',
    MY_UNITS: API_ROOT + '/property_manager/units/mine/',
    SHARED_UNITS: API_ROOT + '/property_manager/shared_units/',
    UNIT_DETAIL: (id) => API_ROOT + `/property_manager/units/${id}/`,
    UNIT_NEAR_ME: (lat, lon, page) =>
      API_ROOT + `/property_manager/units/near-me/${lat},${lon}/?page=${page}`,
    UNITS_PUBLIC: API_ROOT + '/property_manager/units/public/',
    UNIT_SUMMARY: (id) => API_ROOT + `/property_manager/units/${id}/summary/`,
    UNIT_SUMMARY_DETAIL: (id, id2) =>
      API_ROOT + `/property_manager/units/${id}/summary_detail/${id2}/`,
    FILTER_SHARED_UNITS: (field) =>
      API_ROOT + `/property_manager/shared_units/?ordering=${field}`,
    STAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/star/`,
    UNSTAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unstar/`,
    PIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/pin/`,
    UNPIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unpin/`,
    MANAGE_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/`,
  },
  SUB_UNIT: {
    REMOVE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    CREATE_SUB_UNIT: (unitId) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/`,
    MANAGE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    SENSOR_SCAN: (id) =>
      API_ROOT + `/property_manager/stations/${id}/sensor_scan/`,
  },
  SENSOR: {
    DISPLAY: (id) => API_ROOT + `/property_manager/sensors/${id}/display/`,
    DISPLAY_VALUES: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values/`,
    DISPLAY_VALUES_V2: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values_v2/`,
    DISPLAY_HISTORY: (id, hId) =>
      API_ROOT + `/property_manager/sensors/${id}/display_history/`,
    CAMERA_DISPLAY: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/cameras/`,
    REMOTE_CONTROL_OPTIONS: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/remote_control_options/`,
    QUICK_ACTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/quick_action/`,
    CHECK_CONNECTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/check_connection/`,
    UPDATE_SENSOR: (unit_id, station_id, id) =>
      API_ROOT +
      `/property_manager/${unit_id}/sub_units/${station_id}/devices/${id}/`,
  },
  CONFIG: {
    DISPLAY_HISTORY: API_ROOT + '/property_manager/configs/display_history/',
  },
  POWER_CONSUME: {
    DISPLAY_HISTORY:
      API_ROOT + '/property_manager/power_consume/display_history/',
  },
  SHARE: {
    UNITS: API_ROOT + '/property_manager/sharing/units/',
    UNIT_PERMISSIONS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/permissions/`,
    UNITS_MEMBERS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/`,
    UNITS_MEMBER_DETAIL: (id, shareId) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/${shareId}/`,
    SHARE: API_ROOT + '/property_manager/sharing/share/',
  },
  EXTERNAL: {
    WEATHER: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/weather?lat=${location.lat}&lon=${location.lon}` +
      `&appid=${weatherMapId}`,
    UV_INDEX: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/uvi?appid=${weatherMapId}&lat=${location.lat}&lon=${location.lon}`,
    AQI: (location, weatherMapToken) =>
      `https://api.waqi.info/feed/geo:${location.lat};${location.lon}/?token=${weatherMapToken}`,
    GOOGLE_MAP: {
      AUTO_COMPLETE:
        'https://maps.googleapis.com/maps/api/place/autocomplete/json',
      GET_LAT_LNG_BY_PLACE_ID:
        'https://maps.googleapis.com/maps/api/place/details/json',
    },
  },
  PARKING: {
    NEARBY: API_ROOT + '/smart_parking/parking_areas/near_me/',
    NEAREST:
      API_ROOT + '/smart_parking/parking_areas/nearest_available_parking/',
    SAVED_LIST: API_ROOT + '/smart_parking/parking_areas/saved_list/',
    SAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/save/`,
    UNSAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/unsave/`,
    DETAIL: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/`,
    GET_BOOKING_PRICE: (id) =>
      API_ROOT + `/smart_parking/parking_areas/${id}/get_price/`,
    AVAILABLE_TIME_SLOTS: (id) =>
      API_ROOT + `/smart_parking/parkings/${id}/available-time-slots/`,
    PAYMENT_SUCCESS: (id) => API_ROOT + `/smart_parking/payment_success/${id}/`,
    CHECK_CAR_PARKED: API_ROOT + '/smart_parking/spots/check_car_parked/',
    PARKING_INFO: API_ROOT + '/smart_parking/spots/parking_info/',
  },
  CAR: {
    MY_CARS: API_ROOT + '/smart_parking/cars/',
    CHECK_CARS_INFO: API_ROOT + '/smart_parking/cars/check_information/',
    REMOVE_CAR: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE_DEFAULT: (id) =>
      API_ROOT + `/smart_parking/cars/${id}/update_default/`,
  },
  BOOKING: {
    CREATE: API_ROOT + '/smart_parking/bookings/',
    DETAIL: (id) => API_ROOT + `/smart_parking/bookings/${id}/`,
    ACTIVE_SESSION: API_ROOT + '/smart_parking/bookings/active_session/',
    HISTORY: (page) =>
      API_ROOT + `/smart_parking/bookings/history/?page=${page}`,
    SCAN_TO_BOOK: API_ROOT + '/smart_parking/bookings/scan_to_book/',
    SCAN_TO_CONFIRM: API_ROOT + '/smart_parking/bookings/scan_to_confirm/',
    EXTEND_INFO: (id) =>
      API_ROOT + `/smart_parking/bookings/${id}/extend_info/`,
    EXTEND: (id) => API_ROOT + `/smart_parking/bookings/${id}/extend/`,
    CANCEL: (id) => API_ROOT + `/smart_parking/bookings/${id}/cancel/`,
    STOP: (id) => API_ROOT + `/smart_parking/bookings/${id}/stop/`,
  },
  EMERGENCY: {
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
  },
  BILLING: {
    LIST_PAYMENT_METHODS_BY_COUNTRY: (code) =>
      API_ROOT + `/billing/list-methods/${code}/`,
    PAYMENT: {
      STRIPE: {
        PROCESS: (id) => API_ROOT + `/billing/payments/stripe/process/${id}/`,
        ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
        CREATE_PAYMENT_INTENT: (id) =>
          API_ROOT + `/billing/payments/stripe/sca/create-payment/${id}/`,
        PAYMENT_INTENT_SUCCESS: (intent_id) =>
          API_ROOT +
          `/billing/payments/stripe/sca/payment_success/${intent_id}/`,
      },
    },
    DEFAULT_PAYMENT_METHODS: API_ROOT + '/billing/default_payment_method/',
  },
  NOTIFICATION: {
    REGISTER_SIGNAL_ID: API_ROOT + '/accounts/register-signal-id/',
    LIST_ALL_NOTIFICATIONS: (page, type) =>
      API_ROOT + `/notifications/notifications/?page=${page}&type=${type}`,
    SET_READ: (id) => API_ROOT + `/notifications/notifications/${id}/set_read/`,
    NUMBER: API_ROOT + '/notifications/notifications/number/',
  },
  EMERGENCY_BUTTON: {
    CREATE_CONTACT: API_ROOT + '/emergency_button/contacts/',
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
    REMOVE_CONTACTS: (id) => API_ROOT + `/emergency_button/contacts/${id}/`,
    SEND_ALERT: API_ROOT + '/emergency_button/events/',
    NOTIFY_SMS: (id) => API_ROOT + `/emergency_button/events/${id}/notify_sms/`,
    FASLE_ALARM: (id) =>
      API_ROOT + `/emergency_button/events/${id}/false_alarm/`,
    RESOLVE: (id) => API_ROOT + `/emergency_button/events/${id}/resolve/`,
  },
  IOT: {
    CHIP_MANAGER: {
      WATCH_CONFIGS: API_ROOT + '/chip_manager/watch_configs/',
      PUSHER_AUTH: API_ROOT + '/chip_manager/pusher/auth/',
    },
  },
};

export default API;
