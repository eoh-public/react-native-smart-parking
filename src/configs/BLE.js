export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
