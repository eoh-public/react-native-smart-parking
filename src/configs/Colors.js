export const Colors = {
  Primary: '#00979D',
  Text: '#363636',
  Golden: '#b2973f',
  White: '#FFFFFF',
  Red: 'red',
  Green: 'green',
  Gray: '#838383',
  LightGray: '#989898',
  Error: '#F10303',
  DarkLayer: '#00000080',
  LightDarkLayer: '#0000004C',
  LightingDarkLayer: '#00000033',
  LightWhiteLayer: '#F7F7F7D9',
  CoinTagBG: '#F7F7F7',
  Black: '#000000',
  Disabled: '#f8f8f8',
  Border: '#BFBFBF',
  ButtonTag: '#989898',
  WrapGray: '#F5F5F5',
  SwitchButton: '#909090',
  Read: '#F3F3F3',
  Orange: '#F08229',
  BG: '#F0F2F5',
  BGImage: '#F4F8F8',
  Transparent: '#00000000',
  TransparentWhite: '#FFFFFF00',
  BGMapCircle: '#E6F7FF30',
  BGNotRain: '#FEFFE6',
  BorderRain: '#36CFC9',
  Shadow: '#303133',
  ShadownTransparent: 'rgba(48, 49, 51, 0.1)',
  TextTransparent: 'transparent',
  TextGray: 'gray',
  LightOrange: '#FEEFE7',

  //Range Cyan
  Cyan2: '#B5F5EC',
  Cyan2opa30: '#B5F5EC4D',
  Cyan6: '#13C2C2',

  //Range GeekBlue
  GeekBlue3: '#ADC6FF',

  // Range Gray
  Gray4: '#E8E8E8',
  Gray5: '#D9D9D9',
  Gray6: '#BFBFBF',
  Gray7: '#8C8C8C',
  Gray8: '#595959',
  Gray9: '#262626',
  Gray2: '#FAFAFA',
  Gray3: '#F5F5F5',
  Gray11: '#3031331a',
  Gray12: '#DDD',
  Gray13: '#CCC',
  Gray14: '#EEE',
  Gray16: '#333',
  Gray17: '#D3D3D3',
  Gray18: '#9B9A9B',
  Gray19: '#E5E5E5',

  // Range Green
  Green1: '#F6FFED',
  Green3: '#B7EB8F',
  Green6: '#52C41A',
  Green7: '#00979D',
  Green8: '#D9F7BE',
  Green9: '#389E0D',
  Green10: '#237804',

  //Yellow
  Yellow6: '#FADB14',
  Yellow4: '#FFF566',

  //Orange
  Orange3: '#FFD591',
  Orange4: '#FFC069',
  Orange6: '#FA8C16',
  Orange7: '#FAAD14',

  //Red
  Red9: '#A8071A',
  Red8: '#CF1322',
  Red7: '#FA541C',
  Red6: '#F5222D',
  Red5: '#FF4D4F',
  Red4: '#FF7875',
  Red3: '#FFA39E',
  Red2: '#FFCCC7',
  Red1: '#FFF1F0',

  //Purple
  Purple3: '#D3ADF7',
  Purple6: '#722ED1',

  //Pink
  Pink1: '#EFDBFF',

  //Blue
  Blue1: '#F0F5FF',
  Blue2: '#D6E4FF',
  Blue3: '#ADC6FF',
  Blue4: '#85A5FF',
  Blue5: '#597EF7',
  Blue6: '#2F54EB',
  Blue7: '#1D39C4',
  Blue9: '#91D5FF',
  Blue10: '#1890FF',
  Blue11: '#096DD9',
  Blue12: '#0050B3',

  //Range Volcano:
  Volcano3: '#FFBB96',

  //lime
  Lime6: '#A0D911',

  Cyan1: '#E6FFFB',

  LightPrimary: '#EBF7F8',

  BlackTransparent1: 'rgba(0, 0, 0, 0.1)',
  BlackTransparent2: 'rgba(0, 0, 0, 0.2)',
  BlackTransparent3: 'rgba(0, 0, 0, 0.3)',
  BlackTransparent4: 'rgba(0, 0, 0, 0.4)',
  BlackTransparent5: 'rgba(0, 0, 0, 0.5)',
  BlackTransparent6: 'rgba(0, 0, 0, 0.6)',
  BlackTransparent16: 'rgba(0, 0, 0, 0.16)',
  WhiteTransparent8: 'rgba(255, 255, 255, 0.8)',
};
