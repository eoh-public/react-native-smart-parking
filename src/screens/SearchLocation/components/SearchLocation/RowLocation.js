import React, { memo, useCallback } from 'react';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';

import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import { CircleView } from '../../../../commons';
import { roundDecimal } from '../../../../utils/Number';

const RowLocation = memo(({ item, fromApi, onPress }) => {
  const onPressItem = useCallback(() => {
    onPress && onPress(item);
  }, [item, onPress]);
  return (
    <TouchableOpacity onPress={onPressItem}>
      <View style={[styles.rowContainer, styles.paddingHorizontal]}>
        <CircleView
          size={32}
          backgroundColor={Colors.Gray4}
          center
          style={styles.icon}
        >
          <IconOutline
            name={fromApi ? 'environment' : 'history'}
            size={16}
            color={Colors.Gray8}
          />
        </CircleView>
        <Text
          type="H4"
          color={Colors.Gray9}
          style={styles.textTitle}
          numberOfLines={2}
        >
          {item.description}
        </Text>
      </View>
      {!!item.distance_meters && (
        <View style={styles.rowContainerDescription}>
          <Text
            type="Label"
            color={Colors.Gray8}
            style={styles.textDistance}
            numberOfLines={1}
          >
            {item.distance_meters &&
              roundDecimal(item.distance_meters / 1000, 1, 'km')}
          </Text>
          <Text
            type="Label"
            color={Colors.Gray8}
            style={styles.textAddress}
            numberOfLines={2}
          >
            {item.formatted_address}
          </Text>
        </View>
      )}
      <View style={styles.line} />
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  paddingHorizontal: {
    paddingHorizontal: 16,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  rowContainerDescription: {
    marginTop: -12,
    flexDirection: 'row',
  },
  icon: {
    marginVertical: 16,
    marginRight: 24,
  },
  textTitle: {
    flex: 1,
  },
  textDistance: {
    width: 64,
    marginRight: 8,
    textAlign: 'center',
    overflow: 'visible',
  },
  textAddress: {
    flex: 1,
    marginBottom: 16,
  },
  line: {
    position: 'absolute',
    right: 16,
    left: 72,
    bottom: 0,
    height: 1,
    backgroundColor: Colors.Gray4,
  },
});

export default RowLocation;
