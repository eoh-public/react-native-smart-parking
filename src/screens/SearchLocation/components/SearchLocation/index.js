import React, { memo, useCallback, useRef, useState, useEffect } from 'react';
import { StyleSheet, View, TouchableOpacity, TextInput } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import { useNavigation } from '@react-navigation/native';
import { t } from 'i18n-js';

import { Colors, Fonts } from '../../../../configs';
import { TESTID } from '../../../../configs/Constants';

const SearchBarLocation = memo(({ onTextInput }) => {
  const { goBack } = useNavigation();
  const [input, setInput] = useState('');
  const timeOutTextChange = useRef(null);

  const onPressRight = useCallback(() => {
    setInput('');
  }, []);

  useEffect(() => {
    if (timeOutTextChange.current) {
      clearTimeout(timeOutTextChange.current);
    }
    timeOutTextChange.current = setTimeout(
      () => {
        onTextInput && onTextInput(input);
      },
      input ? 300 : 0
    );
  }, [input, timeOutTextChange, onTextInput]);

  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.buttonLeft} onPress={goBack}>
        <IconOutline name="left" size={24} />
      </TouchableOpacity>
      <TextInput
        style={styles.textInput}
        value={input}
        onChangeText={setInput}
        placeholder={t('smart_parking_add_destination')}
        placeholderTextColor={Colors.Gray6}
        underlineColorAndroid={null}
        testID={TESTID.SEARCH_BAR_INPUT}
      />
      <TouchableOpacity
        disabled={!input}
        style={styles.buttonRight}
        onPress={onPressRight}
      >
        <IconOutline
          name={input ? 'close' : 'search'}
          size={24}
          color={input ? Colors.Black : Colors.Gray6}
        />
      </TouchableOpacity>
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    marginVertical: 24,
    marginHorizontal: 16,
    height: 48,
    backgroundColor: Colors.White,
    borderRadius: 24,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    flexDirection: 'row',
    overflow: 'hidden',
  },
  buttonLeft: {
    paddingLeft: 20,
    paddingRight: 10,
    justifyContent: 'center',
  },
  buttonRight: {
    paddingLeft: 10,
    paddingRight: 16,
    justifyContent: 'center',
  },
  textInput: {
    flex: 1,
    height: 48,
    justifyContent: 'center',
    paddingLeft: 10,
    fontFamily: Fonts.Regular,
    fontSize: 14,
    color: Colors.Gray9,
  },
});

export default SearchBarLocation;
