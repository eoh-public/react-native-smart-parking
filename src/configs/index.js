import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
