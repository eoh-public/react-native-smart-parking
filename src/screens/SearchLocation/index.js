/* eslint-disable no-empty */
import React, { useState, useCallback, useEffect } from 'react';
import { StyleSheet, FlatList, SafeAreaView } from 'react-native';
import Config from 'react-native-config';
import { useNavigation } from '@react-navigation/native';
import { t } from 'i18n-js';

import { Colors, API } from '../../configs';
import SearchBarLocation from './components/SearchLocation';
import RowLocation from './components/SearchLocation/RowLocation';
import Text from '../../commons/Text';
import { axiosGet } from '../../utils/Apis/axios';
import { getCurrentLatLng } from '../../utils/CountryUtils';
import useKeyboardShow from '../../hooks/Common/useKeyboardShow';
import AsyncKeys from '../../utils/AsyncKey';

import { getData, storeData } from '../../utils/Storage';
import { removeDuplicateSearch } from '../../utils/Converter/array';

const SearchLocation = ({ route }) => {
  const { onSelectLocation } = route.params;
  const { goBack } = useNavigation();
  const [searchData, setSearchData] = useState([]);
  const [inputSearch, setInputSearch] = useState('');
  const { keyboardBottomPadding } = useKeyboardShow();
  const [recentData, setRecentData] = useState([]);

  const getRecentData = useCallback(async () => {
    let recentSearch = await getData(AsyncKeys.RECENT_SEARCH);
    if (recentSearch === undefined || recentSearch === null) {
      recentSearch = [];
    } else {
      recentSearch = JSON.parse(recentSearch);
    }
    setRecentData(recentSearch);
  }, []);

  useEffect(() => {
    getRecentData();
  }, [getRecentData]);

  const onTextInput = useCallback(
    async (input) => {
      setInputSearch(input);
      if (!input) {
        return;
      }
      const api_key = Config.GOOGLE_MAP_API_KEY;
      try {
        const location = await getCurrentLatLng();
        const config = {
          params: {
            input: input,
            key: api_key,
            sessiontoken: 123456324,
            location: `${location.lat},${location.lng}`,
            radius: 10000,
            origin: `${location.lat},${location.lng}`,
          },
        };

        const { success, data } = await axiosGet(
          API.EXTERNAL.GOOGLE_MAP.AUTO_COMPLETE,
          config
        );
        if (success) {
          setSearchData(data.predictions);
        }
      } catch (error) {}
    },
    [setInputSearch]
  );

  const saveRecentSearch = useCallback(
    async (item) => {
      const { description } = item;
      let recentSearch = recentData;
      removeDuplicateSearch(recentSearch, description);
      recentSearch = [item, ...recentSearch];
      if (recentSearch.length > 10) {
        recentSearch = recentSearch.slice(0, 10);
      }
      await storeData(AsyncKeys.RECENT_SEARCH, JSON.stringify(recentSearch));
    },
    [recentData]
  );

  const onPressRowLocation = useCallback(
    (item) => {
      delete item.distance_meters;
      (async () => {
        await saveRecentSearch(item);
      })();
      onSelectLocation(item);
      goBack();
    },
    [saveRecentSearch, onSelectLocation, goBack]
  );

  const renderItem = useCallback(
    ({ item }) => (
      <RowLocation
        item={item}
        fromApi={!!inputSearch}
        onPress={onPressRowLocation}
      />
    ),
    [inputSearch, onPressRowLocation]
  );
  const keyExtractor = (item) => item.description;

  return (
    <SafeAreaView style={styles.wrap}>
      <SearchBarLocation onTextInput={onTextInput} />
      {!inputSearch && (
        <Text
          semibold
          type="Body"
          color={Colors.Gray9}
          style={styles.paddingHorizontal}
        >
          {t('recent')}
        </Text>
      )}
      <FlatList
        style={styles.wrap}
        data={inputSearch === '' ? recentData : searchData}
        keyExtractor={keyExtractor}
        renderItem={renderItem}
        contentContainerStyle={{
          paddingBottom: keyboardBottomPadding,
        }}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  wrap: {
    flex: 1,
    backgroundColor: Colors.White,
  },
  paddingHorizontal: {
    paddingHorizontal: 16,
  },
});
export default SearchLocation;
