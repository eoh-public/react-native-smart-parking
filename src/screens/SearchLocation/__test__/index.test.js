import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { TextInput } from 'react-native';

import { TESTID } from '../../../configs/Constants';
import SearchLocation from '../index';

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      goBack: jest.fn(),
    }),
  };
});

describe('Test SearchLocation container', () => {
  let tree;
  const route = {
    params: {
      onSelectLocation: jest.fn(),
    },
  };

  test('create SearchLocation', () => {
    act(() => {
      tree = renderer.create(<SearchLocation route={route} />);
    });
    const instance = tree.root;
    const textInput = instance.findAllByType(TextInput);
    const searchBar = textInput.find(
      (item) => item.props.testID === TESTID.SEARCH_BAR_INPUT
    );

    expect(searchBar.props.placeholder).toBe('Tôi muốn đỗ xe gần...');
  });
});
