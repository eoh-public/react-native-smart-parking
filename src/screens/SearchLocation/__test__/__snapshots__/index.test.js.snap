// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchLocation container render SearchLocation container 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 24,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "marginHorizontal": 16,
        "marginVertical": 24,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 20,
          "paddingRight": 10,
        }
      }
      testID="SEARCH_BAR_BUTTON_LEFT"
    >
      <Text
        accessibilityLabel="icon-left"
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Tôi muốn đỗ xe gần..."
      placeholderTextColor="#BFBFBF"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#262626",
          "flex": 1,
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "height": 48,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
      testID="SEARCH_BAR_INPUT"
      underlineColorAndroid={null}
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 16,
        }
      }
      testID="SEARCH_BAR_BUTTON_RIGHT"
    >
      <Text
        accessibilityLabel="icon-search"
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    Gần đây
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 0,
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
